### ISSUE TITLE
Error Occurs When Deploying Firebase Function

### ISSUE THREAD

**Issue by pamalite** [GoogleCloudPlatform/cloud-functions-emulator#86]
### Description

I did the npm install of the emulator from [Google website](https://cloud.google.com/functions/docs/emulator). And, whenever I issue the deploy command like so

`cd myFirebaseWorkDir`
`functions deploy helloWorld --local-path=./functions --trigger-http`

I get the following error:

```
Warning: You're using Node.js v6.10.2 but Google Cloud Functions only supports v6.9.1.
ERROR: Error: Cannot read property '1' of null
    at Request._callback (/usr/local/lib/node_modules/@google-cloud/functions-emulator/node_modules/google-auth-library/lib/transporters.js:100:13)
    at Request.self.callback (/usr/local/lib/node_modules/@google-cloud/functions-emulator/node_modules/request/request.js:188:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/usr/local/lib/node_modules/@google-cloud/functions-emulator/node_modules/request/request.js:1171:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/@google-cloud/functions-emulator/node_modules/request/request.js:1091:12)
    at IncomingMessage.g (events.js:291:16)
    at emitNone (events.js:91:20)
```

I tried `firebase deploy` the function I have and it is working in Firebase. I was able to call it using `curl`. 

Please help. 

Thanks!

### Specifications

- node: 6.10.2
- npm: 3.10.10
- firebase-functions: 0.5.4
- firebase-admin: 4.2.0
- firebase-tools: 3.6.0
- functions: 1.0.0

**Comment by jmdobry** [comment]
Might this be related to #74? What version of the emulator are you running? `functions --version`

**Comment by pamalite** [comment]
Yup. Looks similar. I tried the recommended fix in the thread and I still get the same thing. 

The emulator version as follows:

```
> functions --version
Warning: You're using Node.js v6.10.2 but Google Cloud Functions only supports v6.9.1.
1.0.0-alpha.16
```
Checking against the repo, seems like this is the latest tag.

**Comment by pamalite** [comment]
I managed to solve the problem myself. 

In the `index.js` file, I have the following code snippet:

```
exports.app_do_image = functions.https.onRequest((req, res) => { ... } );
```

I am very new to NodeJS. So, after looking at the examples I can search, for both Firebase and Google Cloud Functions, I saw that the function name has no underscore characters in them. 

I changed the snippet to:

```
exports.appDoImage = functions.https.onRequest(...);
```

After that, when I do `functions deploy`, the emulator works. 

Anyway, the Firebase's own `firebase deploy` works with underscore. I am not sure what is the issue here.
